swaynag/swaynag.c

```diff
@@ -63,12 +39,12 @@ static void swaynag_button_execute(struct swaynag *swaynag,
 			} else if (pid == 0) {
 				// Child of the child. Will be reparented to the init process
 				char *terminal = getenv("TERMINAL");
-				if (button->terminal && terminal && strlen(terminal)) {
+				if (button->terminal && terminal && *terminal) {
 					sway_log(SWAY_DEBUG, "Found $TERMINAL: %s", terminal);
-					if (!terminal_execute(terminal, button->action)) {
-						swaynag_destroy(swaynag);
-						_exit(EXIT_FAILURE);
-					}
+					execlp(terminal, terminal, "-e", "sh", "-c", button->action, NULL);
```

Comment:
Some terminals use `-e cmd arg1 arg2 arg3`, some terminals use `-e "cmd arg1 arg2 arg3"`. The script allowed to handle both.